src/Mover.ts

```diff
@@ -895,13 +895,64 @@ export class MoverAPI implements Types.MoverAPI {
                 ignoreUncontrolled: true,
                 useActiveModalizer: true,
             });
+            if (isGrid) {
+                // finds the last focusable element on the horizontal
+                next = undefined;
+                let cur: HTMLElement | undefined | null = focusable.findPrev({
+                    container,
+                    currentElement: focused,
+                    ignoreUncontrolled: true,
+                    useActiveModalizer: true,
+                });
+                let nextElementX1 = Math.ceil(
+                    cur?.getBoundingClientRect().left ?? 0
+                );
+                while (cur && nextElementX1 < focusedElementX1) {
+                    next = cur;
+                    cur = focusable.findPrev({
+                        container,
+                        currentElement: cur,
+                    });
+                    nextElementX1 = Math.ceil(
+                        cur?.getBoundingClientRect().left ?? -1
+                    );
+                }
+            }
         } else if (keyCode === Keys.End) {
             next = focusable.findLast({
```

Comment:
Let's do `if (isGrid) {} else {}` to avoid unnecessary `findLast()` call. It is a few more symbols, but that function call is not the most lightweight thing in the world.